File: 278-first-bad-version/first-bad-version.c
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

int firstBadVersion(int n) {
    int a = 1, b = n, c;
    while (a < b) {
        c = a + (b - a) / 2;
        if (isBadVersion(c))
            b = c;
        else
            a = c + 1;
    }
    return b;
}

File: 278-first-bad-version/first-bad-version.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        a,b=1,n
        while a<b:
            c=(a+b)//2
            if isBadVersion(c):
                b=c
            else:
                a=c+1
        return b

        